### ISSUE TITLE
Summary : Crash when opening some file variants

### ISSUE THREAD

**Issue by magnesj** [equinor/ecl#550]
When reading some variants of summary files, libecl crashes.

A mail with test data set is sent to @markusdregi 

The call stack looks like this, and for the example case well name / group name is null :

```
	ResInsight.exe!__hash_get_node(const hash_struct * __hash, const char * key, bool abort_on_error) Line 85	C++
 	ResInsight.exe!hash_has_key(const hash_struct * hash, const char * key) Line 535	C++
 	ResInsight.exe!ecl_smspec_install_special_keys(ecl_smspec_struct * ecl_smspec, smspec_node_struct * smspec_node) Line 847	C++
 	ResInsight.exe!ecl_smspec_index_node(ecl_smspec_struct * ecl_smspec, smspec_node_struct * smspec_node) Line 1076	C++
 	ResInsight.exe!ecl_smspec_add_node(ecl_smspec_struct * ecl_smspec, smspec_node_struct * smspec_node) Line 1116	C++
 	ResInsight.exe!ecl_smspec_fread_header(ecl_smspec_struct * ecl_smspec, const char * header_file, bool include_restart) Line 1236	C++
 	ResInsight.exe!ecl_smspec_fread_alloc(const char * header_file, const char * key_join_string, bool include_restart) Line 1266	C++
 	ResInsight.exe!ecl_sum_fread(ecl_sum_struct * ecl_sum, const char * header_file, const stringlist_struct * data_files, bool include_restart, bool lazy_load, int file_options) Line 213	C++
 	ResInsight.exe!ecl_sum_fread_alloc(const char * header_file, const stringlist_struct * data_files, const char * key_join_string, bool include_restart, bool lazy_load, int file_options) Line 271	C++
 	ResInsight.exe!openEclSum(const QString & inHeaderFileName, bool includeRestartFiles) Line 82	C++
 	ResInsight.exe!RifReaderEclipseSummary::open(const QString & headerFileName, bool includeRestartFiles) Line 130	C++

```

**Comment by magnesj** [comment]
See https://github.com/OPM/ResInsight/issues/3892

**Comment by joakim-hove** [comment]
It might not solve the problem - but you should upgrade the libecl distribution as the code i question has been significantly refactored.

**Comment by bjornjensen** [comment]
Tested commit cf05ed6c7b40f7b27cb718b3389ef21345ea4ba1.
The bug is still there.

**Comment by magnesj** [comment]
This will probably be resolved by #615

**Comment by dotfloat** [comment]
Assuming #615 has fixed this.
